spec/controllers/reservations_controller_spec.rb

```diff
@@ -29,8 +30,45 @@
     end
   end
 
+  describe "DELETE #destroy" do
+    context "when user is logged in" do
+      before do
+        sign_in_user
+        delete_destroy(wishlist.id, item.id, reservation.id)
```

Comment:
I checked the tests and they are all passing. But I don't think it's a bug in our authorization logic. Doesn't FactoryGirl just automatically use correct IDs given how `:wishlist`, `:item` and `:reservation` are defined? See below:
`:wishlist` includes a `user` inside:
```ruby
  factory :wishlist do
    name Faker::Lorem.sentence(2)
    description Faker::Lorem.paragraph
    user
  end
```

`:item` includes a `wishlist`:
```ruby
  factory :item do
    name Faker::Lorem.sentence(2)
    description Faker::Lorem.paragraph
    wishlist
  end
``` 
and finally, `:reservation` includes `item`:
```ruby
  factory :reservation do
    item
  end
```
I think this makes FactoryGirl assume correct IDs and provides the "connection" between `:wishlist` <-> `:item` <-> `:reservation` you assumed is missing.